File: nuido_flow/flows/core/mapper_node.py
# THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
#
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT
#
# THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
# DO NOT USE IT IN PRODUCTION.

from .base_node import BaseNode
from ..tools.mapper_tools import map_nested_dict

class MapperNode(BaseNode):
    def _process(self, params):
        super()._process(params)

        dict_map = self.definition["dict_map"]
        res = map_nested_dict(params, dict_map)
        res = res["maproot"]

        return res

File: nuido_flow/static/src/models/core/mapper_node.js
// THIS FILE IS A PART OF PUBLIC REPOSITORY https://github.com/yonitjio/exploring-odoo
//
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT
//
// THIS SOFTWARE IS EXPERIMENTAL AND FOR EDUCATIONAL PURPOSE ONLY.
// DO NOT USE IT IN PRODUCTION.
import { NodeModel } from "@nuido/models/node";
import { Default } from "@nuido/utils/registry";
import { uuidv4 } from "@nuido/utils/utils";
export class MapItem {
    constructor(id, name = "", value = "", parent_id = "") {
        this.id = id;
        this.name = name;
        this.value = value;
        this.parent_id = parent_id;
    }
}
export class MapperNodeModel extends NodeModel {
    setup() {
        const inId = "in-" + this.id + "-1";
        this.addInPort(inId, Default, 1);
        const outId = "out-" + this.id + "-1";
        this.addOutPort(outId, Default, 1);
        this.map = [new MapItem(uuidv4(), "maproot")];
    }
}
